src/views/forms/NewMarketForm/NewMarketSelectionStep.tsx

```diff
@@ -0,0 +1,457 @@
+        message: stringGetter({
+          key: STRING_KEYS.NOT_ENOUGH_BALANCE,
+          params: {
+            NUM_TOKENS_REQUIRED: initialDepositAmount,
+            NATIVE_TOKEN_DENOM: chainTokenLabel,
+          },
+        }),
+      };
+    }
+
+    return null;
+  }, [nativeTokenBalance, stringGetter]);
+
+  useEffect(() => {
+    if (assetToAdd) {
+      setTempLiquidityTier(assetToAdd.liquidityTier);
+      setLiquidityTier(assetToAdd.liquidityTier);
+    }
+  }, [assetToAdd]);
+
+  const filteredPotentialMarkets = useMemo(() => {
+    return potentialMarkets?.filter(
+      ({ baseAsset, numOracles }) =>
+        exchangeConfigs?.[baseAsset] !== undefined &&
+        Number(numOracles) >= NUM_ORACLES_TO_QUALIFY_AS_SAFE &&
+        !marketIds.includes(`${baseAsset}-USD`)
+    );
+  }, [exchangeConfigs, potentialMarkets, marketIds]);
+
+  return (
+    <Styled.Form
+      onSubmit={(e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        if (canModifyLiqTier) {
+          setLiquidityTier(tempLiquidityTier);
+          setCanModifyLiqTier(false);
+        } else {
+          onConfirmMarket();
+        }
+      }}
+    >
+      <h2>
+        {stringGetter({ key: STRING_KEYS.ADD_A_MARKET })}
+        <span>
+          {stringGetter({ key: STRING_KEYS.BALANCE })}:{' '}
+          <Output
+            type={OutputType.Number}
+            value={nativeTokenBalance}
+            fractionDigits={2}
+            slotRight={
+              <Tag style={{ marginTop: '0.25rem', marginLeft: '0.5ch' }}>{chainTokenLabel}</Tag>
+            }
+          />
+        </span>
+      </h2>
+      <SearchSelectMenu
+        items={[
+          {
+            group: 'markets',
+            groupLabel: stringGetter({ key: STRING_KEYS.MARKETS }),
+            items:
+              filteredPotentialMarkets?.map((potentialMarket: PotentialMarketItem) => ({
+                value: potentialMarket.baseAsset,
+                label: potentialMarket?.assetName ?? potentialMarket.baseAsset,
+                tag: `${potentialMarket.baseAsset}-USD`,
+                onSelect: () => {
+                  setAssetToAdd(potentialMarket);
+                },
+              })) ?? [],
+          },
+        ]}
+        label={stringGetter({ key: STRING_KEYS.MARKETS })}
+      >
+        {assetToAdd ? (
+          <Styled.SelectedAsset>
+            {assetToAdd?.assetName ?? assetToAdd.baseAsset} <Tag>{assetToAdd?.baseAsset}-USD</Tag>
+          </Styled.SelectedAsset>
+        ) : (
+          'e.g. "BTC-USD"'
+        )}
+      </SearchSelectMenu>
+      {assetToAdd && (
+        <>
+          <div>{stringGetter({ key: STRING_KEYS.POPULATED_DETAILS })}</div>
+          <div>
+            <Styled.Root value={tempLiquidityTier} onValueChange={setTempLiquidityTier}>
+              <Styled.Header>
+                {stringGetter({ key: STRING_KEYS.LIQUIDITY_TIER })}
+                <Styled.ButtonRow>
+                  <Button
```

Comment:
can we make this a smaller size?

Reply:
Changed the details to `mini-book`

Reply:
Before | After
--- | ---
<img width="604" alt="Screen Shot 2024-01-30 at 11 04 24 AM" src="https://github.com/dydxprotocol/v4-web/assets/13111994/07f07f24-0e78-41c1-b03b-e25dfd33ebb1"> | <img width="601" alt="Screen Shot 2024-01-30 at 11 02 35 AM" src="https://github.com/dydxprotocol/v4-web/assets/13111994/3eb826f5-cc16-4cce-b9c0-234c6d4d607a">